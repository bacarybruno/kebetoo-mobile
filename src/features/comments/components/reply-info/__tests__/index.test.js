import setupTest from '@app/config/jest-setup'

import ReplyInfo, { DeleteIconButton } from '../index'

const givenReplyInfo = setupTest(ReplyInfo)({
  info: {
    content: 'A comment reply',
    author: {
      displayName: 'Bruno',
    },
  },
  size: 20,
  onClose: jest.fn(),
})

it('renders ReplyInfo', () => {
  const { wrapper } = givenReplyInfo()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('closes reply info', () => {
  const { wrapper, props } = givenReplyInfo()
  wrapper.root.findByType(DeleteIconButton).props.onPress()
  expect(props.onClose).toBeCalledTimes(1)
})
