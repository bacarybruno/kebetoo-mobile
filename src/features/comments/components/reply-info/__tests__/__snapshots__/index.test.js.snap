// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReplyInfo 1`] = `
<View
  style={
    Object {
      "height": 20,
      "marginHorizontal": 5.333333333333333,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(229, 229, 234, 0.5)",
          "borderLeftWidth": 3,
          "borderRadius": 5,
          "flexDirection": "row",
          "marginTop": 8,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        },
        Object {
          "borderColor": "#4db6ac",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 16,
            },
            Array [
              Object {
                "color": "rgba(0,0,0,0.54)",
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "marginBottom": 4,
              },
            ],
          ]
        }
      >
        Replying to Bruno
      </Text>
      <Text
        numberOfLines={1}
        style={
          Array [
            false,
            false,
            Object {
              "backgroundColor": "transparent",
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "letterSpacing": -0.24,
              "lineHeight": 20,
            },
            false,
            Array [
              Object {
                "color": "rgba(0,0,0,0.87)",
              },
              Object {
                "backgroundColor": "transparent",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              undefined,
            ],
          ]
        }
      >
        A comment reply
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(229, 229, 234, 255)",
          "borderRadius": 1000,
          "height": 18,
          "justifyContent": "center",
          "opacity": 1,
          "width": 18,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "rgba(0,0,0,0.87)",
            "fontFamily": "Ionicons",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "normal",
          }
        }
        text="ionicon-ios-close"
      />
    </View>
  </View>
</View>
`;
